bodhi/client/__init__.py

```diff
@@ -131,8 +130,13 @@ def query(**kwargs):
 @click.option('--staging', help='Use the staging bodhi instance',
               is_flag=True, default=False)
 def request(update, state, user, password, **kwargs):
-    client = Bodhi2Client(username=user, password=password, staging=kwargs['staging'])
-    resp = client.request(update, state)
+    client = bindings.BodhiClient(username=user, password=password, staging=kwargs['staging'])
+
+    try:
+        resp = client.request(update, state)
+    except bindings.UpdateNotFound as exc:
+        raise click.BadParameter(unicode(exc), param_hint='UPDATE')
```

Comment:
Eventually I do want to convert the codebase to Python 3, but until then I want strings to all be unicode objects in Bodhi since they will be unicode objects in Python 3 as well and since unicode objects are encoding aware. We'll just have to drop the explicit casting of them when we switch.